internal/services/dynatrace/dynatrace_monitors_resource.go

```diff
@@ -0,0 +1,268 @@
+package dynatrace
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dynatrace/2021-09-01/monitors"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type MonitorsResource struct{}
+
+var _ sdk.ResourceWithUpdate = MonitorsResource{}
+
+type MonitorsResourceModel struct {
+	Name                          string            `tfschema:"name"`
+	ResourceGroup                 string            `tfschema:"resource_group_name"`
+	Location                      string            `tfschema:"location"`
+	MonitoringStatus              bool              `tfschema:"monitoring_enabled"`
+	MarketplaceSubscriptionStatus string            `tfschema:"marketplace_subscription"`
+	IdentityType                  string            `tfschema:"identity_type"`
+	PlanData                      []PlanData        `tfschema:"plan"`
+	UserInfo                      []UserInfo        `tfschema:"user"`
+	Tags                          map[string]string `tfschema:"tags"`
+}
+
+type PlanData struct {
+	BillingCycle  string `tfschema:"billing_cycle"`
+	EffectiveDate string `tfschema:"effective_date"`
+	PlanDetails   string `tfschema:"plan"`
+	UsageType     string `tfschema:"usage_type"`
+}
+
+type UserInfo struct {
+	Country      string `tfschema:"country"`
+	EmailAddress string `tfschema:"email"`
+	FirstName    string `tfschema:"first_name"`
+	LastName     string `tfschema:"last_name"`
+	PhoneNumber  string `tfschema:"phone_number"`
+}
+
+func (r MonitorsResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"location": azure.SchemaLocation(),
+
+		"identity_type": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			Default:  "SystemAssigned",
+			ValidateFunc: validation.StringInSlice([]string{
+				"SystemAssigned",
+				"UserAssigned",
+			}, false),
+		},
```

Comment:
we have common Identity Types we should be using here - see github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema